### PR TITLE
Accounts methods

### PR DESCRIPTION
**Pull request by ailox**
This PR implements support for creating, updating and deleting accounts by providing a wrapper arount deluge's `core.get_known_accounts`, `core.create_acccount`, `core.update_account`, and `core.remove_account` RPC methods.

I slightly broke convention by not naming the "RemoveAccount" method "DeleteAccount" (as in "DeleteTorrent"), since I believe the method names should be the same as the one that upstream deluge provides.

### CODE REVIEW

**Comment by ailox** [comment]
I tested this with Deluge 2.0.3 (1.3.x doesn't support multiple accounts).

A crude test to verify that this is working would be:

```go
var newAccount = delugeclient.Account{
	Username:  "alex",
	Password:  "alex",
	AuthLevel: delugeclient.AuthLevelAdmin,
}
var updatedAccount = delugeclient.Account{
	Username:  newAccount.Username,
	Password:  "axel",
	AuthLevel: delugeclient.AuthLevelReadonly,
}

c.CreateAccount(newAccount)
fmt.Println(c.KnownAccounts())
c.UpdateAccount(updatedAccount)
fmt.Println(c.KnownAccounts())
c.RemoveAccount(newAccount.Username)
fmt.Println(c.KnownAccounts())
```

File: methods.go [hunk 1]
```diff
@@ -352,7 +352,7 @@ func (c *Client) RemoveAccount(username string) (bool, error) {
 	var args rencode.List
 	args.Add(username)
 
-	// perform login
+	// remove account
```
Comment (ailox) [inline methods.go]: This is correct now, but it became a little redundant. Should we remove it?

Comment (gdm85) [reply methods.go]: Yes

**Comment by ailox** [comment]
Will try to implement this and test it. Thanks for the extensive review!

**Comment by gdm85** [comment]
@ailox you're right, the `NewAccount` constructor felt wrong. I have changed it as per your original function semantics but preserved the check on length of dictionary values and removed the TODOs.

I also adjusted the return values in case of errors to be more consistent with the rest of the code.

One part which is missing is adding some accounts functionality to the CLI example program, would you like to work on that before the PR is merged? For example you could add only the `KnownAccounts` one.

**Comment by ailox** [comment]
I have not used it so far, but I'll add an list-accounts flag for the cli util.

File: cli/cli.go [hunk 2]
```diff
@@ -173,10 +176,24 @@ func main() {
...
+			fmt.Fprintf(os.Stderr, "ERROR: could not list all accounts: %v\n", err)
+			os.Exit(6)
+		}
+
+		je := json.NewEncoder(os.Stdout)
+		je.SetIndent("", "\t")
+		if err := je.Encode(accounts); err != nil {
 			os.Exit(7)
 		}
-		fmt.Println(string(b))
```
Comment (ailox) [inline cli/cli.go]: streaming the output, instead of writing it into a buffer and printing all at once.

**Comment by ailox** [comment]
I think this is everything that can fit into the scope of this PR, from my point this seems ready to merge.

**Comment by gdm85** [comment]
Thanks for your work and patience in going through the changes @ailox!

**Pull request merged**